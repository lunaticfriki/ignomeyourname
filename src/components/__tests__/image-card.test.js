import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import ImageCard from '../image-card'

describe('Searcher component suite', () => {
  const dummyProps = {
    gn: {},
    setGnome: jest.fn(),
    viewInfo: jest.fn(),
    handleModalInfo: jest.fn(),
  }

  const wrapperMount = mount(<ImageCard {...dummyProps} />)

  test('ImageCard snapshot', () => {
    const component = renderer.create(<ImageCard {...dummyProps} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Should trigger a function to handle modal info', () => {
    wrapperMount.find('article').simulate('click')
    expect(dummyProps.handleModalInfo.mock.calls.length).toBe(1)
  })
})
