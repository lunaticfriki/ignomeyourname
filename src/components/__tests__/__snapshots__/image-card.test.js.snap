// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searcher component suite ImageCard snapshot 1`] = `
<article
  onClick={[Function]}
>
  <p />
  <div
    className="c-image"
  >
    <img />
  </div>
</article>
`;
